### PR TITLE
Add unique validation rule to fields which have a unique constraint

### PR DESCRIPTION
**Pull request by josegonzalez**
Note that because of how the built-in `validateUnique` method works, we cannot attach this validation rule to multiple fields.

Refs cakephp/bake#34

> Making this PR here as I don't have a way to run tests atm.

### CODE REVIEW

File: src/Shell/Task/ModelTask.php [hunk 1]
```diff
@@ -569,6 +570,17 @@ public function fieldValidation($fieldName, array $metaData, $primaryKey)
             ]
         ];
 
+        foreach ($schema->constraints() as $constraint) {
+            $constraint = $schema->constraint($constraint);
+            if (!in_array($fieldName, $constraint['columns']) || count($constraint['columns']) > 1) {
+                continue;
+            }
+
+            if ($constraint['type'] == 'unique') {
```
Comment (dereuromark) [inline src/Shell/Task/ModelTask.php]: ===

Comment (markstory) [inline src/Shell/Task/ModelTask.php]: We have constants for this as well I think.

File: src/Shell/Task/ModelTask.php [hunk 2]
```diff
@@ -569,6 +571,17 @@ public function fieldValidation($fieldName, array $metaData, $primaryKey)
             ]
         ];
 
+        foreach ($schema->constraints() as $constraint) {
+            $constraint = $schema->constraint($constraint);
+            if (!in_array($fieldName, $constraint['columns']) || count($constraint['columns']) > 1) {
+                continue;
+            }
+
```
Comment (markstory) [inline src/Shell/Task/ModelTask.php]: Why only unique indexex with 1 column?

Comment (josegonzalez) [inline src/Shell/Task/ModelTask.php]: It looks like the isUnique rule doesn't work for multiple columns.

Comment (lorenzo) [inline src/Shell/Task/ModelTask.php]: I does work with multiple columns, but extra columns need to be passed as `scope`

**Comment by markstory** [comment]
Merging as this is an incremental improvement over what came before.

**Pull request merged**